src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerRegistrationService.cs

```diff
@@ -134,20 +134,19 @@ public void Reanalyze(Workspace workspace, IIncrementalAnalyzer analyzer, IEnume
                 // no specific projects or documents provided
                 if (projectIds == null && documentIds == null)
                 {
-                    coordinator.Reanalyze(analyzer, workspace.CurrentSolution.Projects.SelectMany(p => p.DocumentIds).ToSet(), highPriority);
+                    coordinator.Reanalyze(analyzer, workspace.CurrentSolution.Projects.Select(p => p.Id).ToSet<object>(), highPriority);
```

Comment:
:memo: There are two things I don't like about this call currently:

1. It passes a set of projects instead of passing `workspace.CurrentSolution`
2. It passes a `HashSet<object>` instead of just `Solution`

There are multiple ways it may be possible to improve the clarity of this code. Here are two options, but other options are also possible:

1. Use three overloads for `Reanalyze`
    * One taking `Solution`
    * One taking either `ProjectId` or `IEnumerable<ProjectId>`
    * One taking either `DocumentId` or `IEnumerable<DocumentId>`
2. Use one `Reanalyze` method, but pass a new value type argument which supports the different scenarios listed for the overloaded methods case.

Reply:
these are internal methods that only used internally. sure I can make 2 hashset that is strongly typed or create intermediate type such as ProjectIdOrDocumentId but not sure whether that is needed.

actual user facing APIs are all strongly typed.
http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/Diagnostics/IDiagnosticAnalyzerService.cs,16
http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/SolutionCrawler/ISolutionCrawlerService.cs,17